### PR TITLE
Add SSIM and PSNR image metrics

### PR DESCRIPTION
**Pull request by harrism**
Adds implementations of peak signal-to-noise ratio (`psnr()`) and structural similarity image metric (`ssim()`) in `fvdb.utils.metrics`.

SSIM uses fused_ssim, which is vendored in using CPM and built into libfvdb. The Python implementation is vendored into ssim.py using cmake and installed in the python wheel/package.

Also adds pytests for these functions.

### CODE REVIEW

File: src/cmake/get_fused_ssim.cmake [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Fetch fused-ssim sources via CPM and build a CMake target from its ext.cpp and CUDA sources
+
+CPMAddPackage(
+    NAME fused_ssim
+    GITHUB_REPOSITORY rahul-goel/fused-ssim
+    GIT_TAG origin/main
```
[the remark this answers is not in the document]
Comment (harrism) [reply src/cmake/get_fused_ssim.cmake]: Good suggestion, done.

File: fvdb/utils/metrics/psnr.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    if reduction not in ("none", "mean", "sum"):
+        raise ValueError("reduction must be one of ('none', 'mean', 'sum')")
+
+    if (noisy_images.shape != ground_truth_images.shape) or (noisy_images.dim() != 4):
+        raise ValueError("Input images must have the same shape and be 4-dimensional with shape (B, C, H, W)")
+
+    mse = torch.mean((noisy_images - ground_truth_images) ** 2, dim=(1, 2, 3))  # [B]
+
+    log_max_value = 2.0 * np.log10(max_value)
```
[the remark this answers is not in the document]
Comment (harrism) [reply fvdb/utils/metrics/psnr.py]: max_value is a Python scalar float -- no gradients. We should not expand it to a tensor. Python floats broadcast cleanly with torch tensors and they are device agnostic.
